os/drivers/serial/serialirq.c

```diff
@@ -256,6 +256,16 @@ void uart_recvchars(FAR uart_dev_t *dev)
 	 */
 
 	if (nbytes) {
+#ifdef CONFIG_PM
+		/* uart_open should set pm_uart_domain_id */
+		DEBUGASSERT(pm_uart_domain_id != -1);
+		/* Suspend board sleep if application is wating for UART data */
+		(void)pm_suspend(pm_uart_domain_id);
+#endif
 		dev->received(dev);
```

Comment:
This will just post a semaphore and not perform the actual read operation. So, pm can still go into sleep if we resume here. pm_resume must be called at the end of uart_read when we know that the app has actually read the data.

Reply:
Done